### ISSUE TITLE
Errors and warnings should be written to stderr

### ISSUE THREAD

**Issue by rshin** [ray-project/ray#1884]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.4.0
- **Python version**: 2.7.12
- **Exact command to reproduce**:

### Describe the problem
Ray writes these messages to stdout. These messages should be written to stderr, if they need to be written at all.

This is especially important for programs that use stdout for their normal output (e.g., it would be impossible to write a Python version of `sort` using Ray).

```
Process STDOUT and STDERR is being redirected to /tmp/raylogs/.
Waiting for redis server at 127.0.0.1:52960 to respond...
Waiting for redis server at 127.0.0.1:49818 to respond...
Starting local scheduler with the following resources: {'GPU': 8, 'CPU': 56}.

======================================================================
View the web UI at http://localhost:8892/notebooks/ray_ui97594.ipynb?token=eb36e6d0dbade0d072f1e62b89f2b710f41ae745cd85c0de
======================================================================

WARNING: Falling back to serializing objects of type <type 'instance'> by using pickle. This may be inefficient.
WARNING: Pickling the class <type 'mtrand.RandomState'> failed, so we are using pickle and only registering the class locally.
WARNING: Serializing objects of type <class 'analysis.io_synthesis.IOSynthesisParameters'> by expanding them as dictionaries of their fields. This behavior may be incorrect in some cases.
WARNING: Serializing objects of type <class 'analysis.semantic_synthesizers.UniformBernoulliModel'> by expanding them as dictionaries of their fields. This behavior may be incorrect in some cases.
WARNING: Falling back to serializing objects of type <type 'classobj'> by using pickle. This may be inefficient.
```

**Comment by robertnishihara** [comment]
What's the standard way to do this? E.g., `print(..., file=sys.stderr)`? Or perhaps use the `warnings` module? Or the `logging` module?

**Comment by robertnishihara** [comment]
Fixed now by #2312.
